cfme/tests/cloud/test_providers.py

```diff
@@ -186,6 +189,27 @@ def vm_ip(cfme_vhd, pwsh_ssh):
         pwsh_ssh.run_command("pwsh {}".format(path_script), timeout=180)
 
 
+@pytest.fixture
+def ec2_provider_with_sts_creds(appliance):
+    assume_role_creds = cfme_data['management_systems']['ec2west']['sts_assume_role']['credentials']
```

Comment:
If you first instantiate the provider object
```suggestion
    assume_role_creds = provider.data.sts_assume_role.credentials
```